### PR TITLE
Automatic CI to fetch, merge, filter tools and create static pages

### PR DESCRIPTION
**Pull request by paulzierep**
Working CI that weekly fetches all tools stepwise and then merges them into all_tools.tsv;
updates the main branch via `git pull -s recursive -X ours` merge strategy.
Tested on my main branch.
The filter and static CI are triggered automatically when all_tools.tsv is changed in the main branch.

### CODE REVIEW

**Comment by paulzierep** [comment]
hang on with the review, I need to rebase https://github.com/paulzierep/galaxy_tool_extractor/tree/main first ...

File: .github/workflows/fetch_all_tools.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      matrix:
+        python-version: [3.8]
+        subset:
+         - repositories01.list
+         - repositories02.list
+         - repositories03.list
+         - repositories04.list
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```
Comment (nsoranzo) [inline .github/workflows/fetch_all_tools.yaml:33]:
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```

File: .github/workflows/fetch_all_tools.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          git pull -s recursive -X ours
+          git push
+
+  fetch-all-tools-merge:
+    runs-on: ubuntu-20.04
+    needs: fetch-all-tools-stepwise
+    name: Fetch all tools merge
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```
Comment (nsoranzo) [inline .github/workflows/fetch_all_tools.yaml:59]:
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```

File: .github/workflows/filter_communities.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+permissions:
+  contents: write
+
+jobs:
+  filter-all-tools:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
```
Comment (nsoranzo) [inline .github/workflows/filter_communities.yaml:26]:
```suggestion
    - uses: actions/checkout@v4
    - uses: actions/setup-python@v5
```

File: .github/workflows/run_tests.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    environment: fetch-tools
+    name: Fetch all tool stepwise
+    strategy:
+      matrix:
+        python-version: [3.8]
+        subset:
+         - repositories02.list
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```
Comment (nsoranzo) [inline .github/workflows/run_tests.yaml:27]:
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```

File: .github/workflows/run_tests.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          git pull -s recursive -X ours
+          git push
+
+  # fetch-all-tools-merge:
+  #   runs-on: ubuntu-20.04
+  #   needs: fetch-all-tools-stepwise
+  #   name: Fetch all tools merge
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #     - uses: actions/setup-python@v4
```
Comment (nsoranzo) [inline .github/workflows/run_tests.yaml:54]:
```suggestion
  #     - uses: actions/checkout@v4
  #     - uses: actions/setup-python@v5
```

**Comment by paulzierep** [comment]
I went the fast way to get this done, so we can finalize the publication, its working with the stats and the instance check now: https://github.com/galaxyproject/galaxy_tool_metadata_extractor/pull/35; @nsoranzo I took your review into account !

**Comment by paulzierep** [comment]
I went the fast way to get this done, so we can finalize the publication, its working with the stats and the instance check now: https://github.com/galaxyproject/galaxy_tool_metadata_extractor/pull/35; @nsoranzo I took your review into account !

**Pull request closed without merging**